File: civitasJAVA/src/civitas/SorpresaIrCarcel.java
package civitas;

import java.util.ArrayList;

/**
 * @author Jesús Navarro Merino
 * @author José Alberto Gómez García
 */

public class SorpresaIrCarcel extends Sorpresa{

    private Tablero tablero;

    public SorpresaIrCarcel(String texto, Tablero _tablero) {
        super(texto);
        tablero=_tablero;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual,todos)){
           informe(actual,todos);
           (todos.get(actual)).encarcelar(tablero.getCarcel());
        }
    }
    
    @Override
    protected void informe (int actual, ArrayList<Jugador> todos){
        super.informe(actual, todos);
    }
    
} //Fin de clase SorpresaIrCarcel

File: civitasRUBY/lib/sorpresa_ir_carcel.rb
# encoding:utf-8

module Civitas
  class SorpresaIrCarcel < Sorpresa
    attr_reader :tablero, :texto
    
    def initialize(texto,_tablero)
      super(texto)
      @tablero=_tablero
    end
    
    def self.constructorSorpresaIrCarcel(texto,tablero)
      new(texto,tablero)
    end
    
    def aplicarAJugador(actual,todos)
      if(jugadorCorrecto(actual, todos))
        informe(actual, todos)
        (todos[actual]).encarcelar(@tablero.numCasillaCarcel)
      end
    end
    
    def informe(actual,todos)
      super(actual,todos)
    end
    
    public_class_method :new
    
    
  end
end
